tests/Service/SiteServiceTest.php

```diff
@@ -107,4 +118,54 @@ public function testSiteEntity(): void
         self::assertSame('TEST_ORG_' . $this->id, $this->service->getSiteOrganization());
         self::assertSame('TEST_AWARDEE_' . $this->id, $this->service->getSiteAwardeeId());
     }
+
+    /**
+     * @dataProvider siteDataProvider
+     */
+    public function testIsValidSite($program, $mayoLinkAccount, $switchSiteName, $checkSiteName, $expectedResult): void
+    {
+        // set env to stable to test isValidSite status
+        $environmentService = $this->createMock(EnvironmentService::class);
+        $environmentService->method('isStable')->willReturn(true);
+        $this->service = new SiteService(
+            static::getContainer()->get(ParameterBagInterface::class),
+            static::getContainer()->get(RequestStack::class),
+            static::getContainer()->get(EntityManagerInterface::class),
+            static::getContainer()->get(UserService::class),
+            $environmentService,
+            static::getContainer()->get(TokenStorageInterface::class),
+        );
+        $this->id = uniqid();
+        $site = $switchSiteName . $this->id;
+        $switchSiteEmail = $site . '@' . self::GROUP_DOMAIN;
+        $this->login('test@example.com', [$site]);
+        $this->service->switchSite($switchSiteEmail);
+        if ($program == User::PROGRAM_NPH) {
+            $this->createNphSite(['mayolinkAccount' => $mayoLinkAccount]);
+        } else {
+            $this->createSite(['mayolinkAccount' => $mayoLinkAccount]);
+        }
+        $checkSiteEmail = $checkSiteName . $this->id . '@' . self::GROUP_DOMAIN;
+        $this->session->set('program', $program);
+        $this->assertEquals($expectedResult, $this->service->isValidSite($checkSiteEmail));
+    }
+
+    public function siteDataProvider(): array
+    {
+        return [
+            // Valid mayolink account number and site email
+            [User::PROGRAM_HPO, '123456789', 'hpo-site-test', 'hpo-site-test', true],
+            // No mayolink account number and with site email
+            [User::PROGRAM_HPO, null, 'hpo-site-test', 'hpo-site-test', false],
+            // No mayolink account number and with invalid site email
+            [User::PROGRAM_HPO, null, 'hpo-site-test', 'hpo-site-test-2', false],
+
+            // Valid mayolink account number and site email
+            [User::PROGRAM_NPH, '123456789', 'nph-site-test', 'nph-site-test', true],
+            // No mayolink account number and with site email
+            [User::PROGRAM_NPH, null, 'nph-site-test', 'nph-site-test', false],
+            // No mayolink account number and with invalid site email
+            [User::PROGRAM_NPH, null, 'nph-site-test', 'nph-site-test-2', false]
```

Comment:
If you make this an associative array, when one of them fails the error report will include the value in the key.  For example: 

```php
return [
    'valid mayolink account number and site email' => [User::PROGRAM_HP), ...]
];
```